File: tiq-e/math/1/main.py
from time import perf_counter
from typing import List


class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        nums = []
        for num in range(1, n + 1):
            if num % 3 == 0 and num % 5 == 0:
                nums.append("FizzBuzz")
            elif num % 3 == 0:
                nums.append("Fizz")
            elif num % 5 == 0:
                nums.append("Buzz")
            else:
                nums.append(str(num))
        return nums


n = 15

start = perf_counter()
result = Solution().fizzBuzz(n)
end = perf_counter()
print("Duration: " + str(end - start))
print("Result: " + str(result))

File: tiq-e/math/1/main.go
package main

import (
	"fmt"
	"strconv"
	"time"
)

func fizzBuzz(n int) []string {
	arr := make([]string, n)
	for i := 0; i < len(arr); i++ {
		val := i + 1
		strVal := strconv.Itoa(val)
		if val%15 == 0 {
			strVal = "FizzBuzz"
		} else if val%3 == 0 {
			strVal = "Fizz"
		} else if val%5 == 0 {
			strVal = "Buzz"
		}
		arr[i] = strVal
	}
	return arr
}

func main() {
	n := 15

	start := time.Now()
	result := fizzBuzz(n)
	end := time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}
